src/opera/pge/dswx_s1/dswx_s1_pge.py

```diff
@@ -441,17 +443,28 @@ def _augment_measured_parameters(self, measured_parameters):
         """
         augmented_parameters = dict()
 
+        descriptions_file = self.runconfig.iso_measured_parameter_descriptions
+
+        if descriptions_file is not None:
+            with open(descriptions_file) as f:
+                descriptions = yaml.safe_load(f)
+
+            missing_description_value = '!Not Found!'
+        else:
+            descriptions = dict()
+            missing_description_value = 'Not Provided'
+
         for name, value in measured_parameters.items():
             if isinstance(value, list):
                 value = json.dumps(value)
 
             data_type = python_type_to_xml_type(value)
 
-            # TODO: This is hardcoded for now, but we should eventually try to guess this
+            # TODO: This is hardcoded for now, but we should eventually try to guess this or add it to the descriptions
+            #  YAML file
             attr_type = 'processingInformation'
 
-            # TODO: Filling this attribute is also on the back burner for now
-            attr_description = f'TBA: Description for {name}'
+            attr_description = descriptions.setdefault(name, dict(description=missing_description_value))['description']
 
             attr_name = (name.title()
```

Comment:
More a note to remind myself, but a good future ticket will be to figure out a good way to refactor this into a separate mapping dictionary that is PGE specific. The code can then just iterate over this dictionary, making the `replace` call for each key/value pair

Reply:
Another idea I had was just leveraging the existing file to include fields that "override" our guesses and derivations in the event we know they'll likely not be fully correct